### PR TITLE
Added support for `@Security` annotations

### PR DESCRIPTION
**Pull request by dragosprotung**
This addresses #1163

### CODE REVIEW

**Comment by dragosprotung** [comment]
@GuilhemN not sure how to fix the 2 broken tests. some guidance will be appreciated

**Comment by GuilhemN** [comment]
I think it's an issue with `json_decode` somewhere in the code, for which we should use the second parameter to convert json objects to php arrays. Or maybe it's an issue in exsyst/swagger...

I can't spend a lot of time on this right now but I'll try to help you more a bit later this the week. Thanks for your PR :)

**Comment by dragosprotung** [comment]
Well, that solved it :)
Ready for review

File: Resources/doc/index.rst [hunk 1]
```diff
@@ -153,6 +154,7 @@ To document your routes, you can use the SwaggerPHP annotations and the
          *     description="The field used to order rewards"
          * )
          * @SWG\Tag(name="rewards")
+         * @Security(name="api_key")
```
Comment (dbu) [inline Resources/doc/index.rst]: can you please mention this in the text that follows, and say a few words what "name" is?

**Comment by dragosprotung** [comment]
@dbu I think this should be called `Security` as it refers to the Security object from the operation (https://swagger.io/specification/#operationObject, https://swagger.io/specification/#securityRequirementObject) and not the security schema (top level)

File: Resources/doc/index.rst (hunk 1 above)
Comment (dragosprotung) [reply Resources/doc/index.rst]: Added some description

**Comment by dragosprotung** [comment]
This belongs here because when using only zircote the usage is a bit different:

```php
/**
 * @SWG\Get(
 *     path="/test",
 *     security={{"Bearer":{}}}
 * )
 */
```

Using this bundle:
```php
/**
 * @SWG\Response(
 *     response=200,
 *     description="Returned on success"
 * )
 * @Security(name="Bearer")
 */
```

File: Resources/doc/index.rst [hunk 2]
```diff
@@ -168,6 +170,14 @@ Use models
 As shown in the example above, the bundle provides the ``@Model`` annotation.
 When you use it, the bundle will deduce your model properties.
 
+Use security
```
Comment (GuilhemN) [inline Resources/doc/index.rst]: Imo this should be in a dedicated article about authentication, it is not needed for basic uses.

Comment (dragosprotung) [reply Resources/doc/index.rst]: True, security is not covered at all. Shall I remove this section ? Adding documentation related to security is for another PR maybe

Comment (dbu) [reply Resources/doc/index.rst]: actually yes, and also yes that its out of scope for this PR. maybe lets create an issue to document the "security" things and paste the explanation you wrote in there?

should the annotation be used at all in the example above, if its a special case only?

Comment (dragosprotung) [reply Resources/doc/index.rst]: For basic usage I think this is not needed. I removed it

File: Describer/SwaggerPhpDescriber.php [hunk 3]
```diff
@@ -141,7 +150,7 @@ private function getAnnotations(): Analysis
 
             foreach ($httpMethods as $httpMethod) {
                 $annotationClass = $operationAnnotations[$httpMethod];
-                $operation = new $annotationClass(['_context' => $context, 'path' => $path, 'value' => $implicitAnnotations, 'tags' => $tags]);
+                $operation = new $annotationClass(['_context' => $context, 'path' => $path, 'value' => $implicitAnnotations, 'tags' => $tags, 'security' => $security]);
```
Comment (GuilhemN) [inline Describer/SwaggerPhpDescriber.php]: should be added only if needed imo. (It seems it's always added to the output right now).

Comment (dragosprotung) [reply Describer/SwaggerPhpDescriber.php]: It's the same with tags. Shall I add a check for both ?

Comment (GuilhemN) [reply Describer/SwaggerPhpDescriber.php]: Sure, should be done as well indeed :)

Comment (dragosprotung) [reply Describer/SwaggerPhpDescriber.php]: done

**Comment by GuilhemN** [comment]
Thank you @dragosprotung!

**Pull request merged**